File: scala/collatz-conjecture/src/main/scala/CollatzConjecture.scala
object CollatzConjecture {
  def steps(num: Int): Option[Int] = {

    if (num <= 0) {
      return None
    }

    var cnt = 0
    var num2 = num
    while (num2 != 1) {

      if (num2 % 2 == 0) {
        num2 /= 2
      } else {
        num2 = 3 * num2 + 1
      }
      cnt += 1
    }

    Some(cnt)

  }

}

File: c/collatz-conjecture/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int num) {

    if (num < 1) return -1;

    int cnt = 0;
    while (num > 1){
        if (num % 2 == 0) {
            num /= 2;
        } else {
            num = 3 * num + 1;
        }
        ++cnt;
    }
    return cnt;
}
